nesta/production/batchables/unsdg/unsdg_text_predict/run.py

```diff
@@ -72,27 +88,54 @@ def run():
     model_bucket = os.environ["BATCHPAR_model_bucket"]
     model_key = os.environ["BATCHPAR_model_key"]
     model_date = os.environ["BATCHPAR_model_date"]
+    dictionary_key = os.environ["BATCHPAR_dictionary_key"]
+    phraser_key = os.environ["BATCHPAR_phraser_key"]
+    topic_model_key = os.environ["BATCHPAR_topic_model_key"]
+    stop_words_key = os.environ["BATCHPAR_stop_words_key"]
+
     es_config = ast.literal_eval(os.environ["BATCHPAR_outinfo"])
-    
+
     ids = retrieve_id_file(ids_bucket, ids_key)
     model = get_pkl_object(model_bucket, model_key)
+    dictionary = get_pkl_object(model_bucket, dictionary_key)
+    bigram_phraser = get_pkl_object(model_bucket, phraser_key)
+    sdg_stop_words = get_pkl_object(model_bucket, stop_words_key)
+    lda_model = LdaModel.load(get_presigned_url(model_bucket, topic_model_key))
+
 
     es = Elasticsearch(es_config['internal_host'], port=es_config['port'], sniff_on_start=True)
-    
+
+    #load stop words and update spacy.nlp vocab
+    #load stopwords
+    spacy_nlp_vocab_update(sdg_stop_words)
+
+
     doc_predictions = []
     for doc_id in ids:
         # query ES for text of record with id
         abstract = get_abstract_by_id(
-                es, 
-                index=es_config['index'], 
-                doc_type=es_config['type'], 
+                es,
+                index=es_config['index'],
+                doc_type=es_config['type'],
                 doc_id=doc_id
                 )
         if abstract is not None:
-        # preprocess text
-            abstract_clean = ' '.join(clean_and_tokenize(abstract, remove_stops=True))
-            # predict SDGs
-            predictions = model.predict([abstract_clean]).tolist()
+            # preprocess text
+            abstract_tokens = word_tokenise(abstract)
+
+            #load bigram model
+
+            tokens_unigrams_bigrams = generate_bigrams(abstract_tokens, bigram_phraser)
+            clean_tokens = clean_bigrams(tokens_unigrams_bigrams, sdg_stop_words)
+
+            #load lda model and dictionary
+            doc_term_vec = [dictionary.doc2bow(doc) for doc in [clean_tokens]]
```

Comment:
We're already looping through the documents, so you don't need to put this back inside a loop. Just apply bag of words conversion directly.
```suggestion
            doc_term_vec = dictionary.doc2bow(clean_tokens)
```